src/createEffectCapableStore.js

```diff
@@ -68,13 +13,67 @@ export const wrapReplaceReducer = store => nextReducer =>
  * @returns {Function} Store factory
  */
 export default createStore => (rootReducer, initialState) => {
-  const store = createStore(enhanceReducer(rootReducer), new AppStateWithEffects(initialState, []));
+  // Reference to the store dispatch function.
+  let dispatch;
+
+  const cleanTask = new Task();
+  const effectQueue = [];
+
+  /**
+   * Takes all of the effects in the closure effect queue, dispatches them,
+   * captures their values and empties the queue.
+   *
+   * @private
+   * @returns {Array<any>} The return values of dispatch.
+   */
+
+  const cleanEffectQueue = () => {
+    const values = effectQueue.map(dispatch);
```

Comment:
I'm not too sure I understand the cascading actions bit. I can definitely change that though.